src/renderer/navigation/helpers.ts

```diff
@@ -20,17 +20,20 @@
  */
 
 import type { LocationDescriptor } from "history";
+import { createPath } from "history";
 import { matchPath, RouteProps } from "react-router";
 import { PageParam, PageParamInit } from "./page-param";
 import { clusterViewRoute, IClusterViewRouteParams } from "../components/cluster-manager/cluster-view.route";
 import { navigation } from "./history";
 
 export function navigate(location: LocationDescriptor) {
-  const currentLocation = navigation.location.pathname;
+  const currentLocation = createPath(navigation.location);
```

Comment:
So this basically produces `/path?and=query#withHash`?

Reply:
I haven't tested `#withHash`, but at least query params should work now.

Reply:
I have tested and it does.